File: Python_Library/Periodic.py
from Media import Media

class Periodic(Media):
	def __init__(self,a,b,c,d,e,f,g,h,i,j,k,l):
		Media.__init__(self,a,b,c,i)
		self.author=d
		self.description=e
		self.publisher=f
		self.publishing_history=g
		self.series=h
		self.related_titles=j
		self.other_forms_of_title=k
		self.govt_doc_number=l
		
	def display(self):
		print("<------------------------------ Record From Periodic --------------------------------------------->\n");
		print("Call Number: "+self.call_number+"\n");
		print("Title: "+self.title+"\n");
		print("Subjects: "+self.subject+"\n");
		print("Author: "+self.author+"\n");
		print("Description: "+self.description+"\n");
		print("Publisher: "+self.publisher+"\n");
		print("Publishing History: "+self.publishing_history+"\n");		
		print("Series: "+self.series+"\n");
		print("Notes: "+self.notes+"\n");
		print("Related Titles: "+self.related_titles+"\n");
		print("Other forms of title: "+self.other_forms_of_title+"\n")
		print("Govt Doc Number: "+self.govt_doc_number+"\n")
		
	def compare_other(self,x):
		if (self.description.find(x) != -1):
			return True
		elif(self.series.find(x) !=-1):
			return True
		elif(self.notes.find(x) !=-1):
			return True
		elif(self.related_titles.find(x) !=-1):
			return True
		else:
			return False
	
 
  

File: ObjectOrientedDesign(LibrarySearch)/Periodic.cpp
#include<iostream>
#include<string>
#include"Periodic.h"
#include"Media.h"
using namespace std;


Periodic :: Periodic( string a,string b,string c,string d, string e, string f, string g, string h, string i,string j,string k,string l) : Media(a, b, c, i)
{  
  this->author=d;
  this->description=e;
  this->publisher=f;
  this->publishing_history=g;
  this->series=h; 
  this->related_titles=j;
  this->other_forms_of_title=k;
  this->govt_doc_number=l;
}

void Periodic::display() const
{
  cout<<"<------------------------------ Record From Periodic File--------------------------------------------->"<<endl;   
  cout<<"Call Number:"<<call_number<<"\n"<<"Title:"<<title<<"\n"<<"Subjects:"<<subject<<"Author:"<<author<<"\n"<<"Description:"<<description<<
  "\n"<<"Publisher:"<<publisher<<"\n"<<"Publishing History:"<<publishing_history<<"\n"<<"Series:"<<series<<"\n"<<"Notes:"<<notes<<"\n"<<"Related Titles:"<<related_titles<<"\n"<<" Other Forms Of Title:"<<other_forms_of_title<<"\n"<<"Govt Doc Number:"<<"\n"<<govt_doc_number<<endl;
}


bool Periodic::compare_other(const string& x) const
{
  if (this->description.find(x) != std::string::npos)
  {
     return true;
  }
   else if(this->series.find(x) != std::string::npos) 
  {
     return true;
  }
   else if(this->notes.find(x) != std::string::npos)
  {
    
     return true;
  } 
  else if(this->related_titles.find(x) != std::string::npos)
  {
     return true;
  }
   else 
  {
     return false;
  }
}


